src/CBLDatabase_Internal.hh

```diff
@@ -352,6 +356,7 @@ private:
 
     litecore::access_lock<Retained<C4Database>> _c4db;
     alloc_slice const                           _dir;
+    CBLEncryptionKey*                           _key;
```

Comment:
Using a bare pointer here is not good practice ... who owns this key? It looks like this points directly to the caller-provided key, so if the app used a stack variable that's going to be overwritten very soon. I think this should be a `CBLEncryptionKey` and copy the key.